### PR TITLE
Fixing benchmark apps

### PR DESCRIPTION
**Pull request by sebastienros**
They should be ignored from the default build logic.

/cc @benaadams

### CODE REVIEW

File: src/Mvc/benchmarkapps/Directory.Build.targets [hunk 1]
```diff
@@ -0,0 +1,2 @@
+<Project>
+</Project>
```
Comment (dougbu) [inline src/Mvc/benchmarkapps/Directory.Build.targets]: Add a comment to the empty files about _not_ importing the root `Directory.Build.*` files

**Review by dougbu — CHANGES_REQUESTED** [review]

File: build/repo.props [hunk 2]
```diff
@@ -57,6 +57,14 @@
                       $(RepositoryRoot)src\ProjectTemplates\Web.ProjectTemplates\content\**\*.csproj;
                       $(RepositoryRoot)src\ProjectTemplates\Web.Spa.ProjectTemplates\content\**\*.csproj;
                       " />
+
+    <!-- Exclude the benchmarks because they use <PackageReference>. -->
+    <ProjectToExclude Include="
```
[the remark this answers is not in the document]
Comment (natemcmaster) [reply build/repo.props]: No, this is a restriction we have given ourselves. See https://github.com/aspnet/AspNetCore/blob/master/docs/ReferenceResolution.md for the thinking on why we decided to do this.

File: src/Servers/Kestrel/perf/PlatformBenchmarks/Directory.Build.props [hunk 3]
```diff
@@ -0,0 +1,5 @@
+<Project>
```
Comment (jkotalik) [inline src/Servers/Kestrel/perf/PlatformBenchmarks/Directory.Build.props]: @sebastienros does this need to be done for the IIS projects too? https://github.com/aspnet/AspNetCore/tree/master/src/Servers/IIS/IIS/benchmarks/IIS.Performance

File: src/Mvc/benchmarkapps/Directory.Build.targets (hunk 1 above)
Comment (sebastienros) [reply src/Mvc/benchmarkapps/Directory.Build.targets]: Done

**Review by dougbu — CHANGES_REQUESTED** [review]

File: src/Mvc/test/Microsoft.AspNetCore.Mvc.FunctionalTests/Microsoft.AspNetCore.Mvc.FunctionalTests.csproj [hunk 4]
```diff
@@ -26,8 +26,6 @@
 
   <ItemGroup>
     <Reference Include="Microsoft.AspNetCore.Mvc.Testing" />
-    <ProjectReference Include="..\..\benchmarkapps\BasicApi\BasicApi.csproj" />
-    <ProjectReference Include="..\..\benchmarkapps\BasicViews\BasicViews.csproj" />
```
[the remark this answers is not in the document]
Comment (benaadams) [reply src/Mvc/test/Microsoft.AspNetCore.Mvc.FunctionalTests/Microsoft.AspNetCore.Mvc.FunctionalTests.csproj]: Currently not detecting that they are broken https://github.com/aspnet/Benchmarks/issues/612

**Review by dougbu — APPROVED** [review]
Discussed offline. Will lose building the benchmark apps in CI as  well as local builds from root. Will also lose the CI tests of the MVC benchmark apps. But, all of this is temporary -- @sebastienros has new automation planned that will catch issues when the apps are run on benchmarks server.

**Pull request merged**